ra2ce/graph/graph_files/graph_files_collection.py

```diff
@@ -0,0 +1,204 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from pathlib import Path
+
+from ra2ce.graph.graph_files.graph_file import GraphFile
+from ra2ce.graph.graph_files.graph_files_enum import GraphFilesEnum
+from ra2ce.graph.graph_files.graph_files_protocol import GraphFileProtocol
+from ra2ce.graph.graph_files.network_file import NetworkFile
+
+
+@dataclass
+class GraphFilesCollection:
+    base_graph: GraphFile = field(
+        default_factory=lambda: GraphFile(
+            default_filename=Path(GraphFilesEnum.BASE_GRAPH.name.lower() + ".p")
+        )
+    )
+    base_graph_hazard: GraphFile = field(
+        default_factory=lambda: GraphFile(
+            default_filename=Path(GraphFilesEnum.BASE_GRAPH_HAZARD.name.lower() + ".p")
+        )
+    )
+    origins_destinations_graph: GraphFile = field(
+        default_factory=lambda: GraphFile(
+            default_filename=Path(
+                GraphFilesEnum.ORIGINS_DESTINATIONS_GRAPH.name.lower() + ".p"
+            )
+        )
+    )
+    origins_destinations_graph_hazard: GraphFile = field(
+        default_factory=lambda: GraphFile(
+            default_filename=Path(
+                GraphFilesEnum.ORIGINS_DESTINATIONS_GRAPH_HAZARD.name.lower() + ".p"
+            )
+        )
+    )
+    base_network: NetworkFile = field(
+        default_factory=lambda: NetworkFile(
+            default_filename=Path(GraphFilesEnum.BASE_NETWORK.name.lower() + ".feather")
+        )
+    )
+    base_network_hazard: NetworkFile = field(
+        default_factory=lambda: NetworkFile(
+            default_filename=Path(
+                GraphFilesEnum.BASE_NETWORK_HAZARD.name.lower() + ".feather"
+            )
+        )
+    )
+
+    @property
+    def _graph_collection(self) -> list[GraphFile | NetworkFile]:
+        return [
+            self.base_graph,
+            self.base_graph_hazard,
+            self.origins_destinations_graph,
+            self.origins_destinations_graph_hazard,
+            self.base_network,
+            self.base_network_hazard,
+        ]
+
+    def has_graphs(self) -> bool:
+        """
+        Tests if any of the types has a graph already read
+
+        Returns:
+            bool: True if any graph already read
+        """
+        return next(
+            iter([x for x in self._graph_collection if x.graph is not None]),
+            False,
+        )
```

Comment:
I notice now that my suggestion was wrong, `next` won't return `True` or `False` but rather the first element that fulfills the condition.
```suggestion
        return any(
            [x for x in self._graph_collection if x is not None]
        )
```
This one is all safe, yet not as optimal as it iterates over the whole collection.

Reply:
Adjusted